app/src/main/java/com/babylon/wallet/android/presentation/accessfactorsources/signatures/GetSignaturesViewModel.kt

```diff
@@ -253,7 +256,11 @@ class GetSignaturesViewModel @Inject constructor(
         signers.forEach { signer ->
             when (val securityState = signer.securityState) {
                 is EntitySecurityState.Unsecured -> {
-                    val factorSourceId = securityState.value.transactionSigning.factorSourceId.asGeneral()
+                    val factorSourceId = when (state.value.signPurpose) {
+                        SignPurpose.SignTransaction -> securityState.value.transactionSigning.factorSourceId.asGeneral()
+                        SignPurpose.SignAuth -> securityState.value.authenticationSigning?.factorSourceId?.asGeneral()
+                            ?: securityState.value.transactionSigning.factorSourceId.asGeneral()
```

Comment:
this is correct for determining the `FactorSourceID`, do you also use this kind of `when` matching for the `DerivationPath` when the private key is (re-)derived?

Reply:
> do you also use this kind of when matching for the DerivationPath when the private key is (re-)derived? 

I am not sure if I understood correctly this.  Are you still talking when signing a transaction or auth?